### DISCUSSION TITLE
Introducing upython-wifi-setup

### DISCUSSION THREAD

**Discussion by shaneapowell** [micropython/micropython#14454] [Show and tell]
# uPython WiFi Setup
https://github.com/shaneapowell/upython-wifi-setup

![aplist](https://github.com/shaneapowell/upython-wifi-setup/raw/main/docs/sc_aplist.png)
![pwd](https://github.com/shaneapowell/upython-wifi-setup/raw/main/docs/sc_password.png)

Like so many of us, getting our iOT projects onto various wifi network easily and seamlessley in a user friendly way is alwasy a challege.  Inspired masively by https://github.com/george-hawkins/micropython-wifi-setup  
I decided that I wanted to create something as flexible and as easily useable, but a little more extensible. 
I also wanted to use an `async` capable web-server like `microdot`.

## Features:
- No pre-compiled web-app.
- Extendable templates
- Able to modify the look of the web-pages, or simply leverage the existing theme in your own project.
- async
- `microdot` (https://github.com/miguelgrinberg/microdot/) web-server for both async and templating engine support.
- `utemplate`(https://github.com/pfalcon/utemplate/tree/master/utemplate) for light-weight templating.
- Easy to get up and running quickly (see readme).

Enjoy!

**Comment by davefes** [comment]
A link to your project?  Doesn't seem to be on your GitHub account.

**Reply by shaneapowell** [reply]
Lol. yeah.. that's a good idea!   oops

**Comment by shaneapowell** [comment]
Update.. added a simple `BLE Nordic UART` mode to this library.  Now, this module can be run with `WiFi AP Captive Portal` mode setup, `BLE UART` mode setup, or both at the same time.   The UART mode has no UI to go with it (yet).  It simply has a basic `JSON` based TX/RX protocol that can be called by your custom android or iOS app (must have BLE support). The app I'm writing with Flutter provides the user with an opiton to provision the wifi now with both modes.  BLE mode by default.  And the necessary instructions to do the CaptivePortal provision.

**Comment by mattytrentini** [comment]
@shaneapowell just thought I'd let you know that we briefly discussed your uPython WiFi Setup project at the [May Melbourne MicroPython Meetup](https://melbournemicropythonmeetup.github.io/May-2024-Meetup/) and there were a few folks keen to try it out... 😄

**Comment by shaneapowell** [comment]
Awesome!  Thanks @mattytrentini . Appreciate that.

I have come across a problem with it just this weekend too. 
My neighbor with an older Samsung phone, when we were trying to get a device onto his wifi, his phone failed to trigger the captive portal when he connected his phone to the AP.  My Pixel 7 detects the captive-portal perfectly.  The same thing happend with an old iPhone 6p that I use as a test device. I think I've narrowed down the issue to something related to a DHCP option (Option 114) that the iPhone is looking for, and I'm guessing so is this older samsung device.  As of yet, I've been unable to find where (if at all) that I can even set this DHCP option to try it out.  I don't even think that the esp-idf framework supports setting this. So.. might be a real issue.  Hence, my addition of this BLE option too.
